mlir/include/mlir/Dialect/Arith/IR/Arith.h

```diff
@@ -154,6 +154,9 @@ Value getReductionOp(AtomicRMWKind op, OpBuilder &builder, Location loc,
                      Value lhs, Value rhs);
 
 arith::CmpIPredicate invertPredicate(arith::CmpIPredicate pred);
+
+/// Creates an `arith.constant` operation with a zero value of type `type`.
+Value getZeroConstant(OpBuilder &builder, Location loc, Type type);
```

Comment:
Should we specify what happens in the failure case? I'm thinking about f8E8M0 specifically which doesn't have a zero value. But there's also memrefs/tensors/ptrs/etc.
I assume this should assert?

Reply:
> `FloatAttr::get(f8E8M0, 0)` wouldn't assert

IMO this is a footgun and I'd rather it asserted...

Reply:
The issue is here:
https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/BuiltinAttributes.td#L654-L657
So while it might be a good decision to reject building the attr with 0. I think that change is out of scope of that PR and requires more discussion, on how to handle attributes for float values that cannot be properly represented.

Reply:
It should be the job `getZeroConstant` or `getZeroAttr` to assert that a zero value can be created IMO. I agree that we shouldn't try to fix the world, but I think that an assert won't hurt here if we can avoid surprising bugs in the future. If you don't think this is the best place for an assertion, we should at least document the failure in the header.

Reply:
I added the assertions for now. I'll create another PR for a proper fix in `FloatAttr`.